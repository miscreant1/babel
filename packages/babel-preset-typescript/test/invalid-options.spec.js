import { transform } from "@babel/core";
import presetTypeScript from "..";

function transformTSWithOptions(source: string, options: any) {
  transform(source, {
    filename: "test.ts",
    presets: [[presetTypeScript, options]],
  });
}
describe("preset-typescript", () => {
  it("should throw when top level options are invalid", () => {
    expect(() => {
      transformTSWithOptions("", { isTsx: true });
    }).toThrowErrorMatchingSnapshot();
  });
  it("should throw when boolean options are not provided with boolean values", () => {
    expect(() => {
      transformTSWithOptions("", { isTSX: "true" });
    }).toThrowErrorMatchingSnapshot();
  });
  it("should throw when string options are not provided with string values", () => {
    expect(() => {
      transformTSWithOptions("", { jsxPragma: 0 });
    }).toThrowErrorMatchingSnapshot();
  });
  it("should throw when isTSX is true but allExtensions is unset", () => {
    expect(() => {
      transformTSWithOptions("", { isTSX: true });
    }).toThrowErrorMatchingSnapshot();
  });
  it("should throw when isTSX is true but allExtensions is false", () => {
    expect(() => {
      transformTSWithOptions("", { isTSX: true, allExtensions: false });
    }).toThrowErrorMatchingSnapshot();
  });
});
