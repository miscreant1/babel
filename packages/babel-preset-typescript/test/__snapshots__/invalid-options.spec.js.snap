// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preset-typescript should throw when boolean options are not provided with boolean values 1`] = `"[BABEL] /Users/jh/code/babel/test.ts: @babel/preset-typescript: 'isTSX' option must be a boolean. (While processing: \\"base$0\\")"`;

exports[`preset-typescript should throw when isTSX is true but allExtensions is false 1`] = `"[BABEL] /Users/jh/code/babel/test.ts: @babel/preset-typescript: isTSX:true requires allExtensions:true (While processing: \\"base$0\\")"`;

exports[`preset-typescript should throw when isTSX is true but allExtensions is unset 1`] = `"[BABEL] /Users/jh/code/babel/test.ts: @babel/preset-typescript: isTSX:true requires allExtensions:true (While processing: \\"base$0\\")"`;

exports[`preset-typescript should throw when string options are not provided with string values 1`] = `"[BABEL] /Users/jh/code/babel/test.ts: @babel/preset-typescript: 'jsxPragma' option must be a string. (While processing: \\"base$0\\")"`;

exports[`preset-typescript should throw when top level options are invalid 1`] = `
"[BABEL] /Users/jh/code/babel/test.ts: @babel/preset-typescript: 'isTsx' is not a valid top-level option.
Maybe you meant to use 'isTSX'? (While processing: \\"base$0\\")"
`;
